packages/authentication/src/providers/auth-strategy.provider.ts

```diff
@@ -0,0 +1,58 @@
+// Copyright IBM Corp. 2019. All Rights Reserved.
+// Node module: @loopback/authentication
+// This file is licensed under the MIT License.
+// License text available at https://opensource.org/licenses/MIT
+
+import {BindingScope, Getter, inject} from '@loopback/context';
+import {
+  extensionPoint,
+  extensions,
+  Provider,
+  ValueOrPromise,
+} from '@loopback/core';
+import {AuthenticationMetadata} from '../decorators/authenticate.decorator';
+import {AuthenticationBindings} from '../keys';
+import {
+  AuthenticationStrategy,
+  AUTHENTICATION_STRATEGY_NOT_FOUND,
+} from '../types';
+
+//this needs to be transient, e.g. for request level context.
+@extensionPoint(
+  AuthenticationBindings.AUTHENTICATION_STRATEGY_EXTENSION_POINT_NAME,
+  {scope: BindingScope.TRANSIENT},
+)
+export class AuthenticationStrategyProvider
+  implements Provider<AuthenticationStrategy | undefined> {
+  constructor(
+    @inject(AuthenticationBindings.METADATA)
+    private metadata: AuthenticationMetadata,
+    @extensions()
+    private authenticationStrategies: Getter<AuthenticationStrategy[]>,
+  ) {}
+  value(): ValueOrPromise<AuthenticationStrategy | undefined> {
+    if (!this.metadata) {
+      return;
+    }
+    const name = this.metadata.strategy;
+
+    return this.findAuthenticationStrategy(name).then(strategy => {
```

Comment:
Let's use `async-await` instead of `then`.

Reply:
I didn't realize I was allowed to : add `async` to the `value()` function of the `Provider` interface, and change its return type from `ValueOrPromise` to `Promise`.